src/main/java/de/tum/in/www1/artemis/web/rest/ExamUserResource.java

```diff
@@ -0,0 +1,114 @@
+import org.springframework.web.multipart.MultipartFile;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.exam.ExamUser;
+import de.tum.in.www1.artemis.repository.*;
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.FileService;
+import de.tum.in.www1.artemis.service.exam.*;
+import de.tum.in.www1.artemis.web.rest.dto.ExamUserDTO;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+/**
+ * REST controller for managing ExamUser.
+ */
+@RestController
+@RequestMapping("api/")
+public class ExamUserResource {
+
+    private final Logger log = LoggerFactory.getLogger(ExamUserResource.class);
+
+    private final UserRepository userRepository;
+
+    private final ExamUserRepository examUserRepository;
+
+    private final FileService fileService;
+
+    private final ExamAccessService examAccessService;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    private final CourseRepository courseRepository;
+
+    private final ExamUserService examUserService;
+
+    public ExamUserResource(ExamUserService examUserService, AuthorizationCheckService authorizationCheckService, UserRepository userRepository, FileService fileService,
+            ExamAccessService examAccessService, ExamUserRepository examUserRepository, CourseRepository courseRepository) {
+        this.userRepository = userRepository;
+        this.fileService = fileService;
+        this.examUserRepository = examUserRepository;
+        this.examAccessService = examAccessService;
+        this.authorizationCheckService = authorizationCheckService;
+        this.courseRepository = courseRepository;
+        this.examUserService = examUserService;
+    }
+
+    /**
+     * POST /courses/:courseId/exams/:examId/exam-users : Update the exam user with the given DTO info
+     *
+     * @param courseId    the id of the course
+     * @param examId      the id of the exam
+     * @param examUserDTO the dto containing exam user info
+     * @return saved examUser ResponseEntity with status 200 (OK) or with status 404 (Not Found)
+     */
+    @PostMapping("courses/{courseId}/exams/{examId}/exam-users")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<ExamUser> updateExamUser(@RequestPart ExamUserDTO examUserDTO, @RequestPart(value = "file", required = false) MultipartFile file,
+            @PathVariable Long courseId, @PathVariable Long examId) {
+        log.debug("REST request to update {} as exam user to exam : {}", examUserDTO.login(), examId);
+
+        examAccessService.checkCourseAndExamAccessForInstructorElseThrow(courseId, examId);
+        var student = userRepository.findOneWithGroupsAndAuthoritiesByLogin(examUserDTO.login())
+                .orElseThrow(() -> new EntityNotFoundException("User with login: \"" + examUserDTO.login() + "\" does not exist"));
```

Comment:
Please pass the id as the second argument to the `EntityNotFoundException` like done elsewhere.

Reply:
Here i don't have an id to pass, but the unique identifier is user login.  So i think its okay not to pass